go/ql/test/experimental/CWE-522-DecompressionBombs/test.go

```diff
@@ -0,0 +1,366 @@
+	zlibklauspost.Read(out) // BAD
+	tarRead = tar.NewReader(zlibklauspost)
+
+	TarDecompressor(tarRead)
+}
+func Zlib(file io.Reader) {
+	var tarRead *tar.Reader
+
+	Zlib, _ := zlib.NewReader(file)
+	var out []byte = make([]byte, 70)
+	Zlib.Read(out) // BAD
+	tarRead = tar.NewReader(Zlib)
+
+	TarDecompressor(tarRead)
+}
+func Snappy(file io.Reader) {
+	var tarRead *tar.Reader
+
+	Snappy := snappy.NewReader(file)
+	var out []byte = make([]byte, 70)
+	Snappy.Read(out)  // BAD
+	Snappy.ReadByte() // BAD
+	tarRead = tar.NewReader(Snappy)
+
+	TarDecompressor(tarRead)
+}
+func SnappyKlauspost(file io.Reader) {
+	var tarRead *tar.Reader
+
+	snappyklauspost := snappyKlauspost.NewReader(file)
+	var out []byte = make([]byte, 70)
+	snappyklauspost.Read(out) // BAD
+	var buf bytes.Buffer
+	snappyklauspost.DecodeConcurrent(&buf, 2) // BAD
+	snappyklauspost.ReadByte()                // BAD
+	tarRead = tar.NewReader(snappyklauspost)
+
+	TarDecompressor(tarRead)
+}
+func S2(file io.Reader) {
+	var tarRead *tar.Reader
+
+	S2 := s2.NewReader(file)
+	var out []byte = make([]byte, 70)
+	S2.Read(out) // BAD
+	var buf bytes.Buffer
+	S2.DecodeConcurrent(&buf, 2) // BAD
+	tarRead = tar.NewReader(S2)
+
+	TarDecompressor(tarRead)
+}
+func Gzip(file io.Reader) {
+	var tarRead *tar.Reader
+
+	gzipRead, _ := gzip.NewReader(file)
+	var out []byte = make([]byte, 70)
+	gzipRead.Read(out) // BAD
+	tarRead = tar.NewReader(gzipRead)
+
+	TarDecompressor(tarRead)
+}
+func GzipKlauspost(file io.Reader) {
+	var tarRead *tar.Reader
+
+	gzipklauspost, _ := gzipKlauspost.NewReader(file)
+	var out []byte = make([]byte, 70)
+	gzipklauspost.Read(out) // BAD
+	var buf bytes.Buffer
+	gzipklauspost.WriteTo(&buf) // BAD
+	tarRead = tar.NewReader(gzipklauspost)
+
+	TarDecompressor(tarRead)
+}
+func PzipKlauspost(file io.Reader) {
+	var tarRead *tar.Reader
+
+	pgzippgzip, _ := pgzipKlauspost.NewReader(file)
+	var out []byte = make([]byte, 70)
+	pgzippgzip.Read(out) // BAD
+	var buf bytes.Buffer
+	pgzippgzip.WriteTo(&buf) // BAD
+	tarRead = tar.NewReader(pgzippgzip)
+
+	TarDecompressor(tarRead)
+}
+func Zstd_Klauspost(file io.Reader) {
+	var tarRead *tar.Reader
+
+	zstd, _ := zstdKlauspost.NewReader(file)
+	var out []byte = make([]byte, 70)
+	zstd.Read(out) // BAD
+	var buf bytes.Buffer
+	zstd.WriteTo(&buf) // BAD
+	var src []byte
+	zstd.DecodeAll(src, nil) // BAD
+	tarRead = tar.NewReader(zstd)
+
+	TarDecompressor(tarRead)
+}
+func Zstd_DataDog(file io.Reader) {
+	var tarRead *tar.Reader
+
+	zstd := zstdDataDog.NewReader(file)
+	var out []byte = make([]byte, 70)
+	zstd.Read(out) // BAD
+	tarRead = tar.NewReader(zstd)
+
+	TarDecompressor(tarRead)
+}
+func Xz(file io.Reader) {
+	var tarRead *tar.Reader
+
+	xzRead, _ := xz.NewReader(file)
+	var out []byte = make([]byte, 70)
+	xzRead.Read(out) // BAD
+	tarRead = tar.NewReader(xzRead)
+
+	TarDecompressor(tarRead)
+}
+
+func TarDecompressor(tarRead *tar.Reader) {
+	var tarOut []byte = make([]byte, 70)
+	tarRead.Read(tarOut) // BAD
```

Comment:
Actually the .expected file doesn't have anything for this line. I'm 90% sure it's because you don't import `io` any more. Try adding a reference to the constant `io.SeekStart` or something, just so it's imported, and see if that fixes it.

Reply:
But it implements the `Read` method of `io.Reader`, doesn't it?